tests/vtpm/conftest.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+import logging
+
+from lib.common import PackageManagerEnum
+
+@pytest.fixture(scope='module')
+def halted_uefi_unix_vm(uefi_vm, unix_vm):
+    assert uefi_vm.is_halted(), "The VM must be halted for these tests"
+    yield uefi_vm
+
+@pytest.fixture(scope='module')
+def snapshotted_halted_uefi_unix_vm(halted_uefi_unix_vm):
+    vm = halted_uefi_unix_vm
+    snapshot = vm.snapshot()
+
+    yield vm
+
+    try:
+        snapshot.revert()
+    finally:
+        snapshot.destroy()
+
+@pytest.fixture(scope='module')
+def unix_vm_with_vtpm(snapshotted_halted_uefi_unix_vm):
+    vm = snapshotted_halted_uefi_unix_vm
+
+    vm.create_vtpm()
+    yield vm
+    # Tear down
+    vm.destroy_vtpm()
+
+@pytest.fixture(scope='module')
+def started_unix_vm_with_vtpm(unix_vm_with_vtpm):
+    vm = unix_vm_with_vtpm
+
+    try:
+        vm.start()
+        vm.wait_for_os_booted()
+    except Exception:
+        vm.shutdown(force=True, verify=True)
+        raise Exception(f"Failed to start VM")
```

Comment:
Moving `wm.start()` out of the try/except block seems enough as `vm.shutdown()` would be executed only if `vm.start() succeeds and `vm.wait_for_os_booted()` fails.

The reason I used `raise` is because while testing this code, nothing was happening upon return from `vm.shutdown()`. Maybe a `pytest.fail()` would leave the original Exception message in log ?

Reply:
We could try (to be tested)